New Drink- Mauna Loa

Sneeker134 opened the pull request:
## About The Pull Request

I've added a spicy drink to the game; the Mauna Loa! Made using capsaicin oil, Kahula, and Bahama Mama.  Uses a sprite I struggled to make, based on the geysers sprite found in Lavaland. Raises body temperature upon drinking, and has a small chance of making the drinker burst into flames. It might not be much, but this is my first PR, and I'm happy just to have managed to make anything! My initial points of things I'm not quite sure about is how high the temperature adjustment should be, and if the sprite I made is really good enough to be in the game (its also my first sprite). I also haven't used the firestacker I saw paired with some of the other uses of IgniteMob, since I wasn't quite sure how it worked, and didn't want to accidentally make my drink super deadly.

The sprite I made:
![mauna_loa](https://user-images.githubusercontent.com/53362182/64925514-346c2380-d7b7-11e9-8f5e-f8148d6c404b.png)

## Why It's Good For The Game

I play bartender quite often, and love mixing the drinks that have other effects besides making you drunk. It adds some variety in the kind of drinks you can make, and I don't think thats ever a bad thing.

## Changelog
:cl:
<!-- add: Added new things
add: Added more things
del: Removed old things
tweak: tweaked a few things
balance: rebalanced something
fix: fixed a few things
soundadd: added a new sound thingy
sounddel: removed an old sound thingy
imageadd: added some icons and images
imagedel: deleted some icons and images
spellcheck: fixed a few typos
code: changed some code
refactor: refactored some code
config: changed some config setting
admin: messed with admin stuff
server: something server ops should know --> add: New fiery drink, Mauna Loa, added to game.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Sneeker134 committed:
Adds my new drink, Mauna Loa!

Sneeker134 committed:
Adds my new drink, Mauna Loa!

Sneeker134 committed:
Merge pull request #3 from Sneeker134/patch-2

Adds my new drink, Mauna Loa!

Sneeker134 committed:
Added the sprite to icons/obj/drinks.dmi

Sneeker134 committed:
Changes temperature change from 15 to 10, to match hellwater.

Fikou commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2092,3 +2092,22 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	if (prob(1))
+		M.IgniteMob()
+	..()
+	..()
```
why do ..() twice

Sneeker134 replied:
Uh, I think thats just a typo. I'll go fix it. Nice catch!

Sneeker134 committed:
Removed redundant ..() call.

spookydonut commented on code/modules/food_and_drinks/recipes/drinks_recipes.dm:
```diff
@@ -781,3 +781,9 @@
+/datum/chemical_reaction/mauna_loa
+	name = "Mauna Loa"
+	id = /datum/reagent/consumable/ethanol/mauna_loa
+	results = list(/datum/reagent/consumable/ethanol/patron = 25)
```
```suggestion
	results = list(/datum/reagent/consumable/ethanol/mauna_loa = 25)
```

Sneeker134 replied:
That seems like a good idea; must've forgot to change some stuff when copy pasting. Thanks!

spookydonut reviewed (changes requested):
might want to make it possible to actually mix the drink

Sneeker134 committed:
Fixed having wrong drink name

Mickyan commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2092,3 +2092,21 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	taste_description = "firey, with an aftertaste of burnt flesh"
+	glass_icon_state = "mauna_loa"
+	glass_name = "Mauna Loa"
+	glass_desc = "Lavaland, in a drink...mug...volanco...thing."
```
volcano

Sneeker134 replied:
Good catch; thanks!

Mickyan commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2092,3 +2092,21 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	description = "Extremely hot; not for the faint of heart!"
+	boozepwr = 40
+	color = "#fe8308" // 254, 131, 8
+	quality = DRINK_GOOD
```
this is pretty hard to make, you could easily bump it up to DRINK_FANTASTIC

Sneeker134 replied:
Ya, that makes sense; honestly just wasn't something I was thinking too much about.

Mickyan commented on code/modules/food_and_drinks/recipes/drinks_recipes.dm:
```diff
@@ -781,3 +781,9 @@
+	name = "Mauna Loa"
+	id = /datum/reagent/consumable/ethanol/mauna_loa
+	results = list(/datum/reagent/consumable/ethanol/mauna_loa = 25)
+	required_reagents = list(/datum/reagent/consumable/capsaicin = 10, /datum/reagent/consumable/ethanol/kahlua = 5, /datum/reagent/consumable/ethanol/bahama_mama = 10)
```
```suggestion
	required_reagents = list(/datum/reagent/consumable/capsaicin = 2, /datum/reagent/consumable/ethanol/kahlua = 1, /datum/reagent/consumable/ethanol/bahama_mama = 2)
```
You should keep the ratios to their simplest form unless you're doing this for a particular reason (do also change the result to 5 units if you change this)

Sneeker134 replied:
I was thinking in button clicks from the Booze machine, not really raw numbers; this makes a lot more sense, and its definitely something I'll try to keep in mind for the future should I make any more drinks!

Sneeker134 committed:
Reducing reagents down to simplest form.

Co-Authored-By: Mickyan <38563876+Mickyan@users.noreply.github.com>

Sneeker134 committed:
Update drinks_recipes.dm

Sneeker134 committed:
Typo fixed and upped drink quality

bobbahbrown commented:
just a side note but the changelog should be descriptive of your pr if/when it gets merged -- so you don't need to have those spellcheck changes in there.

a short and sweet 'add' in the changelog should be good for your pr

Sneeker134 commented:
> just a side note but the changelog should be descriptive of your pr if/when it gets merged -- so you don't need to have those spellcheck changes in there.
> 
> a short and sweet 'add' in the changelog should be good for your pr
> 
> edit: to clarify, it's the text that's actually shown in-game when players see the changelog

ATH1909 commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2092,3 +2092,21 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	// Heats the user up while the reagent is in the body. Occasionally makes you burst into flames.
+	M.adjust_bodytemperature(10 * TEMPERATURE_DAMAGE_COEFFICIENT)
+	if (prob(1))
+		M.IgniteMob()
```
You might wanna slip in an M.adjust_fire_stacks(1) before the M.IgniteMob() here, because IgniteMob() won't set people who don't have any fire stacks on fire.

Maybe make this a prob(5) or even a prob(10) chance (instead of a prob(1) as well?

Sneeker134 replied:
Alright, thanks! I wasn't really sure what firestacks did, so I hoped just using IgniteMob() would just do the trick. Also increased the prob to 5. I don't want the fire to really be that common, but looking at some other procs, it seems like the baseline for unlikely effect to be around there.

ATH1909 commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2092,3 +2092,21 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	boozepwr = 40
+	color = "#fe8308" // 254, 131, 8
+	quality = DRINK_FANTASTIC
+	taste_description = "firey, with an aftertaste of burnt flesh"
```
Generally, taste_descriptions are nouns, not adjectives ("with a hint of firey, with an aftertaste of burnt flesh" sounds weird, no?). Also, you spelled "fiery" wrong. I'd recommend changing this to "fire, with an aftertaste of burnt flesh".

EDIT: Oh, hm, it looks like some other drinks also use adjectives in their descriptions. Go with whatever you want, but I do recommend correcting your spelling of "fiery".

Sneeker134 replied:
Fixed the typo... One of many. I'll try to do better if I submit anything more in the future, I've had a lot of these.

ATH1909 commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2092,3 +2092,21 @@ All effects don't start immediately, but rather get worse over time; the rate is
+	taste_description = "firey, with an aftertaste of burnt flesh"
+	glass_icon_state = "mauna_loa"
+	glass_name = "Mauna Loa"
+	glass_desc = "Lavaland, in a drink...mug...volcano...thing."
```
Remove the comma and add a space after each ellipsis.

Sneeker134 replied:
Fixed typo.

ATH1909 commented on code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:
```diff
@@ -2092,3 +2092,21 @@ All effects don't start immediately, but rather get worse over time; the rate is
+
+/datum/reagent/consumable/ethanol/mauna_loa/on_mob_life(mob/living/carbon/M)
+	// Heats the user up while the reagent is in the body. Occasionally makes you burst into flames.
+	M.adjust_bodytemperature(10 * TEMPERATURE_DAMAGE_COEFFICIENT)
```
Just so you know, this heats you up less than coffee does (unless you're close to or above BODYTEMP_NORMAL). You might wanna bump up the heating rate here.

Sneeker134 replied:
Alright, I wasn't exactly sure how much heating was enough. I took a glimpse at capsaicin oil and took a number, but I guess I severely underestimated how quickly the bonus would tally up over time. I've upped it to 25, though if that too high, I can easily adjust it down; I just don't really have much context yet for how much that it.

ATH1909 reviewed (commented):
I like this PR, but there are a few minor things about it that should probably be adjusted/fixed.

Sneeker134 committed:
Spelling fixes, debuffs buffed

Even more typos pointed out (dang, I suck at English) and fixed, added firestacks so that the mob properly is set on fire on the proc and increased the probability to 5 from 1, increased temperature change from 10 to 25.
